include/deal.II/base/vectorization.h

```diff
@@ -39,6 +39,22 @@
 // 'check_01_cpu_features.cmake', ensures that these feature are not only
 // present in the compilation unit but also working properly.
 
+#if DEAL_II_COMPILER_VECTORIZATION_LEVEL >= 1 && not defined(__SSE2__)
+#error "Mismatch in vectorization capabilities"
+#error "SSE2 was detected during configuration but it is not available in the compilation unit"
```

Comment:
I think this may be hard to decipher if that happens to you because "configuration" and "compilation unit" may not be time points the user is clear about. How about "SSE2 was detected during configuration at deal.II and switched on, but it is apparently not available for the file you are trying to compile at the moment."